$MirOS: ports/mailnews/pine/patches/patch-pine_send_c,v 1.2 2005/05/04 23:37:41 tg Exp $
--- pine/send.c.orig	Mon Sep 12 22:04:03 2005
+++ pine/send.c	Fri Oct  7 19:12:56 2005
@@ -3247,7 +3247,7 @@ pine_send(outgoing, body, editor_title, 
 
     /* NOTE: initial cursor position set below */
 
-    dprint(9, (debugfile, "flags: %x\n", pbf->pine_flags));
+    dprint(9, (debugfile, "flags: %lx\n", pbf->pine_flags));
 
     /*
      * When user runs compose and the current folder is a newsgroup,
@@ -4588,7 +4588,7 @@ pine_send(outgoing, body, editor_title, 
 
 		    pf_uid->writehdr  = 1;
 		    pf_uid->localcopy = 1;
-		    sprintf(uidbuf, "(%d %s)(%d %lu %s)%s",
+		    sprintf(uidbuf, "(%d %s)(%ld %lu %s)%s",
 			    strlen(reply->prefix), reply->prefix,
 			    i, reply->data.uid.validity,
 			    tmp_20k_buf, reply->mailbox);
